src/net.h

```diff
@@ -232,6 +232,7 @@ struct LocalServiceInfo {
     uint16_t nPort;
 };
 
+extern Mutex g_mutex_message_handler;
```

Comment:
The reasoning isn't that this is particularly "good", it's that it's documenting the safety mechanisms we already have. It would have caught https://github.com/bitcoin/bitcoin/pull/21528#discussion_r601317221 for instance.

net_processing is fundamentally coupled with net -- it implements the NetEventsInterface defined and called from net. I mean I guess if you preferred, it could be `NetEventsInterface::m_mutex_message_handler`?

Reply:
And if you're doing that then perhaps you can make it a `PeerManagerImpl::m_mutex_message_handler` and take it in `ProcessMessages()` and `SendMessages()`. Taking that lock once at the top of those functions would be very low cost (since there would never be any contention).

Reply:
I've done this now -- the first three commits introduce `m_mutex_messages`, remove `cs_sendProcessing` and move extra txns to be protected by `m_mutex_messages` instead of `g_cs_orphans`. I think it would be good to cherry-pick those commits into #21186 so that the moved addr variables can actually be marked as guarded by something.

Reply:
Also see #21563